cmd/node/node.go

```diff
@@ -422,6 +428,29 @@ func runNode(ctx context.Context, nc *config) (_ io.Closer, retErr error) {
 	}
 	defer func() { retErr = closeIfErrorf(st, retErr, "failed to close state") }()
 
+	if nc.enableBlockchainUpdatesPlugin {
+		go func() {
+			for {
+				var patch []proto.DataEntry
+				select {
+				case dataEntryKeys := <-bUpdatesExtension.BuPatchRequestChannel:
+					for _, dataEntryKey := range dataEntryKeys {
+						recipient := proto.NewRecipientFromAddress(bUpdatesExtension.L2ContractAddress())
+						dataEntry, err := st.RetrieveEntry(recipient, dataEntryKey)
+						if err != nil {
+							dataEntry = &proto.DeleteDataEntry{Key: dataEntryKey}
+						}
+						patch = append(patch, dataEntry)
+					}
+					bUpdatesExtension.WriteBUPatch(patch)
+				case <-bUpdatesExtension.Ctx.Done():
+					bUpdatesExtension.Close()
```

Comment:
Only writer should close a channel, though it's applicable here because channel closing logic is placed inside. But a bit scary